Bmx.CommandLine/IniConfiguration.cs

```diff
@@ -0,0 +1,81 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using System.Security;
+using Microsoft.Extensions.Configuration;
+
+namespace Bmx.CommandLine {
+	public class IniConfiguration : IBmxConfig {
+		public string? Org { get; }
+		public string? User { get; }
+		public string? Account { get; }
+		public string? Role { get; }
+		public string? Profile { get; }
+
+		public IniConfiguration() {
+			var config = GetConfiguration();
+
+			Org = config["org"];
+			User = config["user"];
+			Account = config["account"];
+			Role = config["role"];
+			Profile = config["profile"];
+		}
+
+		private static IConfiguration GetConfiguration() {
+			// Main config is at ~/.bmx/config
+			var configLocation =
+				Path.Join( Environment.GetFolderPath( Environment.SpecialFolder.UserProfile ), ".bmx", "config" );
+
+			// If set, we recursively look up from CWD (all the way to root) for additional bmx config files (labelled as .bmx)
+			var shouldReadProjectConfig = ShouldReadProjectConfigs( configLocation );
+			string? projectConfigPath = null;
+
+			if( shouldReadProjectConfig ) {
+				projectConfigPath = GetProjectConfigPath();
+			}
+
+			var config = new ConfigurationBuilder().AddIniFile( configLocation );
```

Comment:
`new ConfigurationBuilder().AddIniFile` is called twice in this block (once inside `ShouldReadProjectConfigs`). I feel it can be avoided with better clarity.
Also, if you use `GetValue<bool>`, I don't find `ShouldReadProjectConfigs` adds much value as a separate method.

```suggestion
			var config = new ConfigurationBuilder().AddIniFile( configLocation, optional: true );
			bool shouldReadProjectConfig = config.Build().GetValue<bool>( "allow_project_configs" );
			string? projectConfigPath = null;

			if( shouldReadProjectConfig ) {
				projectConfigPath = GetProjectConfigPath();
			}
```

Reply:
Fair enough, removed it. However I still need to build the configuration twice. First to read it in order to see if project configs should be included, and then again including the project config.